torch/csrc/jit/passes/python_print.cpp

```diff
@@ -753,10 +829,19 @@ struct PythonPrintPass {
 
     // current graph is used to de-dup names within a single graph
     scanBlock(graph.block());
-    assignValuesToTheirUniqueNames(graph.inputs());
+
+    // last param_names.size() arguments to the graph are parameters and not
+    // actual inpputs, we will print these as, e.g. self.foo.bar
+    // while we print the true_inputs out as parameters
+    auto true_inputs = graph.inputs().slice(0, graph.inputs().size() - param_names.size());
```

Comment:
This code feels brittle to me, but it looks like this matches the nn.Module api?

Reply:
it matches how Method works. The contract here is that the last param_names inputs are considered parameters. If we change module then we will have to change this pretty printing code as well.